### ISSUE TITLE
Training on GPU

### ISSUE THREAD

**Issue by groadabike** [borchero/pycave#1]
Hi, I've been trying to train a 1000 components GMM using pycave. My data is a numpy array with shape(2300000, 40).
When I try to train on GPU, the system raise an error and the training crush. However, if I train on CPU, I have no problems.

Are there some requirements to train on the GPU?

A snippet of my code

```
from pycave.bayes import GMM
from torch.utils.data import DataLoader, Dataset

class MyDataset(Dataset):
    def __init__(self, data):
        self.data = data

    def __getitem__(self, index):
        x = torch.from_numpy(self.data[index])
        return x
    
    def __len__(self):
        return len(self.data)

data_set = MyDataset(data) # np array shape (2000000, 40)
loader = DataLoader(data_set, batch_size=10, shuffle=True, num_workers=5)

gmm = GMM(num_components=50, num_features=40, covariance='diag')
history = gmm.fit(loader, gpu=False)
```

From the snippet above, I can't set gpu=True, the program crush

**Comment by borchero** [comment]
Can you maybe provide me with the stacktrace?

**Comment by groadabike** [comment]
Thank you for your response, I don't have stacktrace, the system just crush.
However, I manage to make it work.
I didn't try before but sending the model to GPU first and the fit fixed the problem. ie.:
instead of 
```
gmm = GMM(num_components=50, num_features=40, covariance='diag')
history = gmm.fit(loader, gpu=True)
```
do
```
gmm = GMM(num_components=50, num_features=40, covariance='diag')
gmm = gmm.to('cuda')
history = gmm.fit(loader, gpu=True)
```

**Comment by borchero** [comment]
Great to hear that you got it to work! The model should move itself to the GPU, however, so I'll keep the issue open until I have time to fix this.

**Comment by borchero** [comment]
As I tried to debug this, I could not reproduce the issue with the code you provided. The root cause of the error might have been caused by the `from_numpy` function as it can easily result in a wrong `dtype`.

I'm going to close this for now but feel free to reopen it if you encounter the issue again and can't get rid of the `gmm.to('cuda')` call.

**Comment by groadabike** [comment]
Thank you @borchero for look into this.
No, I can't get rid of `gmm.to('cuda')`.

In case it helps, I'm using:
Ubuntu 20.04.1
Python 3.8.6
pyblaze 2.2.0
pycave  2.0.2
torch 1.7.1
numpy 1.19.2

cheers

**Comment by borchero** [comment]
This should be finally resolved in `2.0.3` :)

**Comment by groadabike** [comment]
Hi, 
Yes, I did upgrade pycave and pyblaze and the issue was resolve.
Cheers
